core/jobs/batch_jobs/manual_voice_artist_name_job_test.py

```diff
@@ -0,0 +1,890 @@
+    VoiceArtistMetadataModelsTestsBaseClass):
+
+    JOB_CLASS: Type[
+        manual_voice_artist_name_job.CreateExplorationVoiceArtistLinkModelsJob
+    ] = (
+        manual_voice_artist_name_job.CreateExplorationVoiceArtistLinkModelsJob
+    )
+
+    def test_empty_storage(self) -> None:
+        self.assert_job_output_is_empty()
+
+    def test_version_is_added_after_running_job(self) -> None:
+        self._create_curated_explorations()
+        self._create_non_curated_exploration()
+
+        job_result_template = (
+            'Generated exploration voice artist link for '
+            'exploration %s.'
+        )
+        self.assert_job_output_is([
+            job_run_result.JobRunResult(
+                stdout=job_result_template % self.CURATED_EXPLORATION_ID_1,
+                stderr=''),
+            job_run_result.JobRunResult(
+                stdout=job_result_template % self.CURATED_EXPLORATION_ID_2,
+                stderr='')
+        ])
+
+        expected_exp_id_to_content_id_to_voiceovers_mapping = {
+            self.CURATED_EXPLORATION_ID_1: {
+                'content_0': {
+                    'en': (self.editor_id_1, self.voiceover_dict_1),
+                    'hi': (self.editor_id_2, self.voiceover_dict_2)
+                },
+                'ca_placeholder_2': {
+                    'en': (self.editor_id_1, self.voiceover_dict_3)
+                },
+                'default_outcome_1': {}
+            },
+            self.CURATED_EXPLORATION_ID_2: {
+                'content_0': {
+                    'en': (self.editor_id_4, self.voiceover_dict_6),
+                },
+                'ca_placeholder_2': {
+                    'en': (self.editor_id_1, self.voiceover_dict_4)
+                },
+                'default_outcome_1': {
+                    'en': (self.editor_id_1, self.voiceover_dict_7),
+                }
+            }
+        }
+
+        exploration_voice_artist_link_models: Sequence[
+            voiceover_models.ExplorationVoiceArtistsLinkModel] = (
+                voiceover_models.ExplorationVoiceArtistsLinkModel.
+                get_all().fetch()
+            )
+
+        for exp_link_model in exploration_voice_artist_link_models:
+            exp_id = exp_link_model.id
+            content_id_to_voiceovers_mapping = (
+                exp_link_model.content_id_to_voiceovers_mapping)
+
+            expected_content_id_voiceovers_mapping = (
+                expected_exp_id_to_content_id_to_voiceovers_mapping[exp_id])
+
+            for content_id, lang_code_to_voiceover_mapping in (
+                    content_id_to_voiceovers_mapping.items()):
+
+                assert isinstance(expected_content_id_voiceovers_mapping, dict)
+
+                self.assertIn(
+                    content_id,
+                    list(expected_content_id_voiceovers_mapping.keys())
+                )
+
+                for lang_code, voiceover_tuple in (
```

Comment:
Could you not just do an equality comparison here, rather than a for loop? A for loop tests inclusion one way but doesn't test exclusion, and a straightforward equality check would be stronger.

Reply:
Done